Allow customer ops to recompute

Hahu803 opened the pull request:
Differential Revision: D85683496

Lucaskabela commented on torch/_functorch/partitioners.py:
```diff
@@ -276,6 +276,14 @@ def _is_primal(node: fx.Node) -> bool:
 def _is_tangent(node: fx.Node) -> bool:
     return node.op == "placeholder" and "tangents" in str(node.target)
 
+def _is_customer(node: fx.Node) -> bool:
```
Additional thought - maybe we need to gate this behind a config? Otherwise we are recomputing all custom triton kernel wrappers which may be expensive

Lucaskabela reviewed (changes requested):
Thanks for minimizing! I think the content looks good know, but we may want to be even more selective with the naming here

Lucaskabela reviewed (commented):
Thanks for the change! I think we need to lint but generally lgtm - will hold off to let @bdhirsh  @zou3519 give final approval

Hahu803 committed:
Allow customer ops to recompute (#167083)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/167083

Differential Revision: D85683496
